ui/src/app/views/admin/worker-model-pattern/worker-model-pattern.component.ts

```diff
@@ -0,0 +1,32 @@
+import {Component} from '@angular/core';
+import {WorkerModelService} from '../../../service/worker-model/worker-model.service';
+import {Table} from '../../../shared/table/table';
+import { ModelPattern } from 'app/model/worker-model.model';
+import {finalize} from 'rxjs/operators';
+
+@Component({
+    selector: 'app-worker-model-pattern',
+    templateUrl: './worker-model-pattern.html',
+    styleUrls: ['./worker-model-pattern.scss']
+})
+export class WorkerModelPatternComponent extends Table {
+
+    workerModelPatterns: Array<ModelPattern> = [];
+    filter: string;
+    loading = false;
+
+    constructor(private _workerModelService: WorkerModelService) {
+        super();
+        this.loading = true;
+        this._workerModelService.getWorkerModelPatterns()
+            .pipe(finalize(() => this.loading = false))
+            .subscribe(wmp => this.workerModelPatterns = wmp);
+    }
+
+    getData(): any[] {
+        if (!this.filter) {
+            return this.workerModelPatterns;
+        }
+        return this.workerModelPatterns.filter((wmp) => wmp.name.toLowerCase().indexOf(this.filter.toLowerCase()) !== -1);
```

Comment:
the filter should filter on type too

Reply:
no we just filter the view via the name (because the type is displayed in the list view)